doc/nrf/protocols/matter/end_product/watchdog.rst

```diff
@@ -33,10 +33,28 @@ The feature is enabled by default for the release build type in all Matter sampl
 To set the timeout for the watchdog timer, configure the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_TIMEOUT` with a value in milliseconds.
 By default, the timeout is set to 10 seconds.
 
+.. _ug_matter_device_watchdog_pause_mode:
+
+Matter watchdog pause mode
+**************************
+
+You can set the Matter watchdog to pause when the CPU is in idle (sleep) mode or when the CPU is halted by the debugger.
+This means the watchdog timer will stop counting down when the CPU is idle or halted.
+
+If pause mode is disabled, the watchdog must be fed within a specified time window, regardless of the CPU state, otherwise the device will reboot.
+If pause mode is enabled, the device will only reboot if the task feeding the watchdog is blocked for longer than the set time window.
+Pause mode is less accurate and may cause the device to reboot later than expected, but it allows feeding the watchdog only when the CPU is active.
+Non-pause mode requires periodic feeding to avoid a timeout, impacting power consumption as the CPU must wake up periodically.
+
+To enable pause mode while CPU is in the idle state, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_IN_SLEEP` Kconfig option to ``y``.
+To enable pause mode during debugging, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_ON_DEBUG` Kconfig option to ``y``.
```

Comment:
this looks to be documentation for customers yet these Kconfigs are for samples, not really seeing how this is useful to customers unless I've missed something?

Reply:
But that's what I mean, these are Kconfigs for those samples, so if I am a customer reading this with my custom application, what use is it if I can't use these Kconfigs because they don't exist for my application?

Reply:
Hmm, Customers in their custom application's Kconfig file must source `"${ZEPHYR_NRF_MODULE_DIR}/samples/matter/common/src/Kconfig"` to get all our configs, so those configs will exist in their application, right?

Reply:
> Hmm, Customers in their custom application's Kconfig file must source `"${ZEPHYR_NRF_MODULE_DIR}/samples/matter/common/src/Kconfig"` to get all our configs, so those configs will exist in their application, right?

No no no, absolutely not, never source sample files

Reply:
So where should I keep kconfigs like those?

Reply:
Ok, I see. The Matter watchdog is just an example/suggestion of how users can use this module in their application and it is not part of Matter. Therefore the kconfigs from here are dedicated just to samples because we need to show how users can implement it in their applications. I will discuss it with the Team and try to change the documentation.

Reply:
@nordicjm I've added a note to the documentation as we discussed above. Do you think that it is enough?